### DISCUSSION TITLE
Does go-sqlcmd truncate nvarchar(max)?

### DISCUSSION THREAD

**Discussion by SirParser** [microsoft/go-sqlcmd#521] [Q&A]
In the old sqlcmd it is necesary to use scripting variable SQLCMDMAXVARTYPEWIDTH or -y switch to allow "max" variables to store strings longer than 256 chars. 
How does it work in the new version? Is the content of nvarchar(max) truncated (e.g. JSON string)? If so, are there settings to avoid that?

**Comment by shueybubbles** [comment]
we've preserved backward compatibility for the most part.  Does the `:XML` command give you the untruncated rendering you need?
